catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/QueryTransformer.java

```diff
@@ -0,0 +1,121 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformer.impl;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.codice.ddf.catalog.ui.metacard.workspace.QueryMetacardImpl;
+import org.codice.ddf.catalog.ui.query.transform.QueryBasicValueTransformer;
+import org.codice.ddf.catalog.ui.query.transform.QuerySortsValueTransformer;
+import org.codice.ddf.catalog.ui.query.transform.QueryValueTransformer;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryTransformer {
+
+  private EndpointUtil util;
+
+  private List<QueryValueTransformer> queryValueTransformers;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(QueryTransformer.class);
+
+  public QueryTransformer(EndpointUtil util) {
+    this.util = util;
+
+    // The filter for each value transformer could match multiple fields. The order of these value
+    // transformers matters.
+    this.queryValueTransformers =
+        Arrays.asList(new QuerySortsValueTransformer(), new QueryBasicValueTransformer());
+  }
+
+  public Map<String, Object> transformMetacardIntoMap(Metacard metacard) {
+    Map<String, Object> map = util.getMetacardMap(metacard);
+    List<Map> sorts =
+        Optional.of(map.remove("sorts"))
+            .filter(value -> value instanceof List)
+            .map(List.class::cast)
+            .map(this::mapSorts)
+            .orElse(Collections.emptyList());
+
+    Map<String, Object> filteredMap =
+        map.entrySet()
+            .stream()
+            .filter(entry -> Objects.nonNull(entry.getValue()))
+            .filter(entry -> !(entry.getValue() instanceof List) || notEmptyList(entry.getValue()))
+            .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+    filteredMap.put("sorts", sorts);
+
+    return filteredMap;
+  }
+
+  private boolean notEmptyList(Object value) {
+    return value instanceof List && !((List) value).isEmpty();
+  }
+
+  public Metacard transformJsonIntoMetacard(Map<String, Object> json) {
+    QueryMetacardImpl query = new QueryMetacardImpl();
+
+    json.forEach(
+        (key, value) -> {
```

Comment:
I let the individual value transformers handle the creation of the attribute so that they can be serialized into whatever format they want. This is mostly to support how the workspace transformer used to serialize these values